Framework/Geometry/src/Objects/MeshObject.cpp

```diff
@@ -22,11 +22,14 @@ using Kernel::Quat;
 /**
 *  Default constuctor
 */
-MeshObject::MeshObject() : MeshObject("") {}
+MeshObject::MeshObject() {} // Should never be called
 
-MeshObject::MeshObject(const std::string &string)
-    : m_boundingBox(), m_object_number(0), m_handler(), m_string(string),
-      m_id(), m_material(), m_triangles(), m_vertices() {}
+MeshObject::MeshObject(const std::vector<size_t> &faces,
+                       const std::vector<V3D> &vertices)
+    : m_boundingBox(), m_object_number(0), m_id(), m_material(),
+      m_triangles(faces), m_vertices(vertices) {
```

Comment:
👍 for your changes....

This is a bit advanced, but as it stands, a `MeshObject` contains a copy of the vertices and faces provided via the constructor arguments. That could be quite expensive. One solution could be to declare both constructor arguments as rvalue references, which allows you to safely`move` (steal) them into the held member vectors. To do this you would have to decalre the Constructor as `MeshObject(const std::vector<size_t> &&faces, std::vector<V3D> &&vertices)`

Come and talk to me if this doesn't make sense to you.

Reply:
I've come across `&&name` in a course I recently did.

Reply:
Would this place any restrictions on the caller of the constructor?

Reply:
Yes, they would have to pass a temporary or it won't compile. In most cases I think they would want to do this.

Reply:
By temporary do you mean something created on the stack (without a new() )?

Reply:
I've tried it and it was OK in the unit test after I used `std::move` on the constructor arguments, but in `MeshObject` I create a scaled version of the `this` to calculate the solid angle with scale factor and the fails for the triangles and I can't fix it with `std::move` or a temporary copy of the triangles.

Reply:
@KarlPalmen I'm a bit tied up this morning. Could you please ask one of the experienced members of the team in the office such as @martyngigg to have a look at this with you?

Reply:
After consulting Martyn, I made two constructors; A flexible constructor that works the old way and an efficient constructor that works the way you suggested.